pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -748,6 +804,16 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1beta1.Pip
 	return nil
 }
 
+// setFinallyStartedTimeIfNeeded sets the PipelineRun.Status.FinallyStartedTime to the current time if it's nil.
+func (c *Reconciler) setFinallyStartedTimeIfNeeded(pr *v1beta1.PipelineRun, facts *resources.PipelineRunFacts) {
+	if pr.Status.FinallyStartTime == nil {
+		pr.Status.FinallyStartTime = &metav1.Time{Time: c.Clock.Now()}
+	}
+	if facts.FinallyStartTime == nil {
+		facts.FinallyStartTime = &pr.Status.FinallyStartTime.Time
+	}
+}
```

Comment:
can we be in a state where `facts.FinallyStartTime` is not nil but `pr.Status.FinallyStartTime` was nil? can those values could ever be different? if so, what's the source of truth (maybe facts because pipelinerun status can be modified)? or could we have this value recorded at one point only?

Reply:
I can't think of a scenario where we could be - the nil checks are just to be perfectly safe.